src/components/carousel/ThumbnailPagination.tsx

```diff
@@ -79,9 +77,12 @@ const ThumbnailPagination: FC<Props> = ({
 
   useEffect(() => {
     if (!paginationCarousel) return;
-    onScroll();
```

Comment:
The logic for evaluating `paginationCarousel.slidesNotInView()` has been re-written to utilise intersection observer. For this reason, `onScroll` call here doesn't correctly evaluate `slidesNotInView` anymore. In order to mitigate this, the callback is now invoked on both `scroll` (existing) and `slidesInView` (new) events.